src/libYARP_OS/include/yarp/os/FixedSizeBuffersManager.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_OS_FIXEDSIZEBUFFERSMANAGER_H
+#define YARP_OS_FIXEDSIZEBUFFERSMANAGER_H
+
+#include <yarp/os/Mutex.h>
+#include <vector>
+#include <stdexcept>
+
+
+// Defined in this file:
+namespace yarp { namespace os { template <typename T> class FixedSizeBuffersManager; }}
+namespace yarp { namespace os { template <typename T> class Buffer; }}
```

Comment:
You can remove these 2 forward declarations, you don't need them

Reply:
sorry, you are wrong.
We need them because `FixedSizeBufferManager' is a friend class of Buffer.

Reply:
You are right ! Yet another reason why I don't like `friend` classes :grimacing: